File: SwissO.Droid/OverviewButton.cs
using Android.Content;
using Android.Util;
using Android.Views;
using Google.Android.Material.Button;
using static SwissO.MyResources;

namespace SwissO.Droid {
    class OverviewButton : MaterialButton {

        private Event.UriArt uriArt;
        private AppManager appManager;
        private Event e;
        private StringResource text;

        public OverviewButton(Context context, IAttributeSet attrs) : base(context, attrs) {
            Click += (sender, e) => {
                appManager.OpenEventDetails(this.e, uriArt);
            };
        }

        public void Init(AppManager appManager, StringResource text, Event e, Event.UriArt uriArt) {
            this.uriArt = uriArt;
            this.appManager = appManager;
            this.e = e;
            this.text = text;
            if (e != null && e.GetUri(uriArt) != null) {
                MyResources_A res = new MyResources_A(Resources);
                SetText(res.GetStringId(this.text));
            }
            SetVisible();
        }

        public void SetVisible() {
            Visibility = e != null && e.GetUri(uriArt) != null ? ViewStates.Visible : ViewStates.Gone;
        }

    }
}

File: app/src/main/java/ch/swisso/OverviewButton.java
package ch.swisso;

import android.content.Context;
import android.util.AttributeSet;

import androidx.annotation.NonNull;

import com.google.android.material.button.MaterialButton;

public class OverviewButton extends MaterialButton {

    public OverviewButton(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    public void init(MyActivity act, int text, @NonNull Event e, Event.UriArt uriArt) {
        boolean hasUri = e.getUri(uriArt) != null;
        setText(text);
        setVisibility(hasUri ? VISIBLE : GONE);
        setOnClickListener(v -> act.openEventDetails(e, uriArt));
    }
}
